packages/sol-tracing-utils/src/trace_collection_subprovider.ts

```diff
@@ -20,6 +20,24 @@ export interface TraceCollectionSubproviderConfig {
     shouldCollectGasEstimateTraces: boolean;
 }
 
+type AsyncFunc = (...args: any[]) => Promise<void>;
+
+// This wrapper outputs errors to console even if the promise gets ignored
+// we need this because web3-provider-engine does not handle promises in
+// the after function of next(after).
+function logErrors(fn: AsyncFunc): AsyncFunc {
+    async function wrappedAsync(...args: any[]): Promise<void> {
+        try {
+            await fn(...args);
+        } catch (e) {
+            // tslint:disable-next-line no-console
+            console.error(e);
+            throw e;
+        }
+    }
+    return wrappedAsync;
+}
```

Comment:
This hack is going in? I was hoping we would come up with something better :)

Reply:
You're absolutely right. I've figured out how to fix it.

Reply:
Oops, I was wrong and I from the past was right. Not about the fact that we shouldn't handle this but about the fact that there is no elegant way to do so.